webnn/resources/test_data/slice.json

```diff
@@ -0,0 +1,772 @@
+            -58.46095416849133,
+            79.80571087632629,
+            -9.857854207596304,
+            42.66519880465225
+          ],
+          "type": "float32"
+        }
+      },
+      "starts": [-1, -2, -1, -1], // [-1, -2, -1, -1] is equal to [1, 0, 2, 1]
+      "sizes": [1, 2, 1, 1],
+      "expected": {
+        "name": "output",
+        "shape": [1, 2, 1, 1],
+        "data": [
+          -27.306041717529297,
+          42.665199279785156
+        ],
+        "type": "float32"
+      }
+    },
+    {
+      "name": "slice float32 4D tensor sizes having special value of -1",
+      "inputs": {
+        "input": {
+          "shape": [2, 2, 3, 2],
+          "data": [
+            28.846251144212147,
+            97.9541470229301,
+            -68.15961736262523,
+            14.978987588557175,
+            90.23090084798065,
+            76.59095547712559,
+            -24.556316258204532,
+            79.58749302462488,
+            65.21375982234204,
+            57.43970862960305,
+            74.41775256222849,
+            -4.513182026141791,
+            0.5424534704526991,
+            80.44634130202977,
+            28.327650022124686,
+            74.0261938866893,
+            -74.54558964005362,
+            -27.30604081001799,
+            -70.4277433212161,
+            59.82631931755259,
+            -58.46095416849133,
+            79.80571087632629,
+            -9.857854207596304,
+            42.66519880465225
+          ],
+          "type": "float32"
+        }
+      },
+      "starts": [1, 0, 2, 1],
+      "sizes": [1, -1, 1, -1], // [1, -1, 1, -1] is equal to [1, 2, 1, 1]
+      "expected": {
+        "name": "output",
+        "shape": [1, 2, 1, 1],
+        "data": [
+          -27.306041717529297,
+          42.665199279785156
+        ],
+        "type": "float32"
+      }
+    },
+    {
+      "name": "slice float32 5D tensor sizes having special value of -1",
+      "inputs": {
+        "input": {
+          "shape": [2, 2, 3, 2, 1],
+          "data": [
+            28.846251144212147,
+            97.9541470229301,
+            -68.15961736262523,
+            14.978987588557175,
+            90.23090084798065,
+            76.59095547712559,
+            -24.556316258204532,
+            79.58749302462488,
+            65.21375982234204,
+            57.43970862960305,
+            74.41775256222849,
+            -4.513182026141791,
+            0.5424534704526991,
+            80.44634130202977,
+            28.327650022124686,
+            74.0261938866893,
+            -74.54558964005362,
+            -27.30604081001799,
+            -70.4277433212161,
+            59.82631931755259,
+            -58.46095416849133,
+            79.80571087632629,
+            -9.857854207596304,
+            42.66519880465225
+          ],
+          "type": "float32"
+        }
+      },
+      "starts": [1, 0, 2, 1, 0],
+      "sizes": [-1, 2, -1, 1, -1], // [-1, 2, -1, 1, -1] is equal to [1, 2, 1, 1, 1]
```

Comment:
Is -2 a valid size? Like I said above, every framework I know of has the ability to accept negative end values, so you can easily crop off the last N elements, but WebNN does not appear to have that, requiring the caller to instead resolve the size beforehand, and yet the start coordinate does accept negative values which are relative, and the inconsistency here doesn't make sense 🤔.